tests/test_target_encoder.py

```diff
@@ -112,3 +112,67 @@ def test_HandleUnknownValue_HaveUnknownInTest_ExpectMean(self):
         obtained = enc.transform(test, test_target)
 
         self.assertEqual(.6, list(obtained['color'])[0])
+
```

Comment:
and maybe another test for multi-level hierarchies where we'd expect an error.

Reply:
Do you mean a test to check it errors elegantly?

Reply:
yes, just an `self.assertRaises` statement that checks for the error of mulit-level hierarchies

Reply:
I just saw you've already implemented that. Very nice!